src/screens/WebsocketResearchScreen/SampleChatScreen.tsx

```diff
@@ -48,54 +53,72 @@ export const styles = StyleSheet.create({
   const updatedChats = updateChatContinuity(chats);
+  const [loading, setLoading] = React.useState(false);
+  const {anonProfileId} = useProfileHook();
+
+  const ownerChat = selectedChannel?.rawJson?.channel?.members?.find(
+    (item: any) => item.user_id === anonProfileId
+  );
+  const memberChat = selectedChannel?.rawJson?.channel?.members?.find(
+    (item: any) => item.user_id !== anonProfileId
+  );
 
   const renderChatItem = React.useCallback(({item, index}) => {
     return <BaseChatItem type={ANONYMOUS} item={item} index={index} />;
   }, []);
 
-  const scrollToEnd = () => {
-    flatlistRef.current?.scrollToEnd();
-  };
+  React.useEffect(() => {
+    return () => {
+      clearReplyPreview();
+    };
+  }, []);
 
   return (
-    <View style={styles.keyboardAvoidingView}>
-      {selectedChannel ? (
-        <ChatDetailHeader
-          onAvatarPress={goToChatInfoScreen}
-          onBackPress={goBackFromChatScreen}
-          onThreeDotPress={goToChatInfoScreen}
-          avatar={selectedChannel?.channelPicture}
-          user={selectedChannel?.name}
-          type={ANONYMOUS}
+    <ScrollContext.Provider value={scrollContext}>
+      <StatusBar barStyle="light-content" backgroundColor={colors.anon_primary} />
+
+      <View style={styles.keyboardAvoidingView}>
+        {selectedChannel ? (
+          <ChatDetailHeader
+            onAvatarPress={goToChatInfoScreen}
+            onBackPress={goBackFromChatScreen}
+            onThreeDotPress={goToChatInfoScreen}
+            avatar={selectedChannel?.channelPicture}
+            user={selectedChannel?.name}
+            type={ANONYMOUS}
+          />
+        ) : null}
+        <FlatList
+          ref={scrollRef}
+          contentContainerStyle={styles.flatlistContainer}
+          style={styles.chatContainer}
+          data={updatedChats}
+          inverted={true}
+          initialNumToRender={20}
+          alwaysBounceVertical={false}
+          keyExtractor={(item, index) => item?.id || index.toString()}
+          renderItem={renderChatItem}
         />
-      ) : null}
-      <FlatList
-        contentContainerStyle={styles.contentContainerStyle}
-        style={styles.chatContainer}
-        data={updatedChats}
-        inverted={true}
-        initialNumToRender={20}
-        alwaysBounceVertical={false}
-        bounces={false}
-        onLayout={scrollToEnd}
-        keyExtractor={(item, index) => item?.id || index.toString()}
-        renderItem={renderChatItem}
-      />
-      <View style={styles.inputContainer}>
-        <AnonymousInputMessage onSendButtonClicked={sendChat} type={ANONYMOUS} />
+
+        {replyPreview && <ChatReplyPreview type={ANONYMOUS} />}
+        <View style={styles.inputContainer}>
+          <AnonymousInputMessage onSendButtonClicked={sendChat} type={ANONYMOUS} />
+        </View>
+        <Loading visible={loading} />
       </View>
-    </View>
+    </ScrollContext.Provider>
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes in this section are mostly related to rendering components and passing props. It's worth noting that the `scrollToEnd` functionality has been removed. If the intention was to automatically scroll to the end of the chat when the component mounts, you might want to reconsider this change.


<!-- This is an auto-generated comment by OSS CodeRabbit -->